### PR TITLE
Update Mode.plot_component

### PR DESCRIPTION
**Pull request by mpasson**
Plotting 'abs' func with field component 'x' or 'y' shows some problems #170.

This is my attempt to solve it. Please fell free to disregard if I missed something in the fix.
Best,

### CODE REVIEW

File: femwell/maxwell/waveguide.py [hunk 1]
```diff
@@ -329,8 +329,9 @@ def plot_component(
 
         if component == "x" or component == "y":
             plot_basis = mfield_t_basis.with_element(ElementVector(ElementDG(ElementTriP1())))
-            mfield_xy = plot_basis.project(mfield_t_basis.interpolate(conv_func(mfield_t)))
+            mfield_xy = plot_basis.project(mfield_t_basis.interpolate(mfield_t))
```
Comment (HelgeGehring) [inline femwell/maxwell/waveguide.py:332]:
```suggestion
            mfield_xy = plot_basis.project(conv_func(mfield_t_basis.interpolate(mfield_t)))
```
would this solve the problem?

**Review by HelgeGehring — COMMENTED** [review]
Thanks a lot!

I think this solves the problem, but I'm not sure if that's actually the clean way.
It works I think as we projected on linear lagrange which makes projection and applying such a function almost exchangable

I would somehow expect (especially for anything higher order), that we'd need to
- interpolate -> apply the function -> project
That way we'd need to apply it to the values at the interpolation points instead of on the degrees of freedom :thinking: 

I made a proposal, but I think it's a little more complicated - the function would need to be applied on each component (once we interpolated, we have a three dimensional vector (with another dimension over the quadrature points)) - don't know if numpy would do it like this :thinking: 

Could you give it a try?

I.e. apply the function somewhere between interpolate and project?

(otherwise, we could also merge this as it's better than what we do right now. But I'd guess we're close to having it completely correct)

File: femwell/maxwell/waveguide.py (hunk 1 above)
Comment (mpasson) [reply femwell/maxwell/waveguide.py]: Thanks for the the quick reply!

I update the code according to your suggestion and it seems to work. 
Let me know if something else is needed.

Comment (HelgeGehring) [reply femwell/maxwell/waveguide.py]: perfect, thanks!

**Comment by HelgeGehring** [comment]
I'll merge once the CICD sucessfully ran

**Pull request merged**